rs/sns/governance/src/governance.rs

```diff
@@ -2007,6 +2017,95 @@ impl Governance {
             .unwrap_or(u64::MAX);
     }
 
+    pub async fn get_sns_statistics(
+        &self,
+        request: GetSnsStatusRequest,
+    ) -> Result<GetSnsStatusResponse, GovernanceError> {
+        let num_recent_proposals = self.recent_proposals(request.time_window_seconds.unwrap());
+        let last_transaction_timestamp = self.get_most_recent_tx_ts().await?;
+
+        // transaction timestamps are in nanoseconds
+        Ok(GetSnsStatusResponse {
+            num_recent_proposals: Some(num_recent_proposals),
+            last_transaction_timestamp: Some(last_transaction_timestamp),
+        })
+    }
+
+    async fn get_most_recent_tx_ts(&self) -> Result<u64, GovernanceError> {
+        // Make the first call to get the current block number
+        let call_icrc3_get_blocks = |args: Vec<GetBlocksRequest>| async {
+            let result =
+                self.ledger
+                    .icrc3_get_blocks(args)
+                    .await
+                    .map_err(|nervous_system_error| {
+                        GovernanceError::new_with_message(
+                            ErrorType::External,
+                            nervous_system_error.error_message,
+                        )
+                    })?;
+
+            Ok::<GetBlocksResult, GovernanceError>(result)
+        };
+        let args = vec![GetBlocksRequest::default()];
+
+        // Make the second call to the last added block to fetch the most
+        // recent transaction.
+        let last_block_number = call_icrc3_get_blocks(args)
+            .await
+            .map(|blocks| blocks.log_length - Nat::from(1_u32))?;
+
+        let args = vec![GetBlocksRequest {
+            start: last_block_number,
+            length: Nat::from(1_u32),
+        }];
+
+        let last_block = call_icrc3_get_blocks(args).await?;
+
+        // TODO asserting/logging if blocks.len() != 1
+        // We assume in each block we have 1 and only 1 transaction.
+        // Block timestamps are in nano seconds
+        let ts =
+            Self::get_block_ts(&last_block.blocks[0].block).unwrap() / Nat::from(ONE_SEC_NANOSEC);
+
+        // As there is no clean way of converting candid::Nat to u64,
+        // we first convert it to a string, remove `_` delimiters, and parse it
+        // to get u64.
+        Ok(ts
+            .to_string()
+            .chars()
+            .filter(|c| *c != '_')
+            .collect::<String>()
+            .parse::<u64>()
+            .unwrap())
+    }
+
+    fn get_block_ts(block: &ICRC3Value) -> Option<Nat> {
```

Comment:
This seems too low-level for putting it into this file, let's add another file / submodule that would include SNS ledger helpers?

Reply:
We could also give move verbose names to our internal functions, e.g., 

```suggestion
    fn get_block_timestamp_nanos(block: &ICRC3Value) -> Option<Nat> {
```

Reply:
Done.